### DISCUSSION TITLE
Stuttering audio transmission over A2DP bluetooth from computer to ESP32

### DISCUSSION THREAD

**Discussion by jason-a-miller2004** [pschatzmann/arduino-audio-tools#799] [Q&A] [answered]
Hey there everyone,

I am trying to do some audio processing. I am playing a sound from my laptop, and sending it to an ESP32 (Adafruit Feather HUZZAH32) using A2DP. I just tried to implement a simple program where I streamed from bluetooth and copied over the signal to serial. The output below is running the following program with a C sine wave drone playing. I am running the same audio and code, but am somehow getting different results. 

I seem to be getting a pure sine wave like I would expect about half of the time, and the rest of the time it's these spikes. I was hoping to get some help/advice on getting more consistent output. So I can start to process the data. ![image](https://user-images.githubusercontent.com/55771862/233897718-ff61f228-b7ee-4db2-9790-8140598a3917.png)`

![image](https://user-images.githubusercontent.com/55771862/233897792-9c764ec9-5c0f-4bc5-8a4c-1cab6298696f.png) `#include "AudioTools.h"
#include "AudioLibs/AudioA2DP.h"

char NAME[] = "Adafruit Feather";
A2DPStream a2dp;
CsvStream<int16_t> out(Serial); 
int sample_rate = 44100;
int channels = 2;

// Arduino Setup
void setup(void) {  
  // Open Serial 
  Serial.begin(115200);
  auto config = out.defaultConfig();
  config.sample_rate = sample_rate; 
  config.channels = channels;
  config.bits_per_sample = sizeof(int16_t)*8; 
  out.begin(config);
  AudioLogger::instance().begin(Serial, AudioLogger::Warning);
  a2dp.begin(RX_MODE, NAME);
}
                             
StreamCopy copier(out, a2dp);                             
void loop() {
  copier.copy();
}

**Comment by pschatzmann** [comment, marked as answer]
I don't understand what you mean with 'C sine wave drone'

You obviously need to find out first where the problem is coming from:
- did you try your sketch with playing music a mobile phone ?
- Maybe the problem is with the os of your audio source. If you use windows try to exchange it with linux or osx
- Maybe you have a bug in the sound generation.

In general I recommend to use the basic API over the Stream based A2DP. There is much less overhead!

I don't think there is a bug in my code, so I convert this to a discussion.

**Comment by pschatzmann** [comment]
And of cause my general recommendation if something is not working still apply: https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working
